src/components/TapBar.tsx

```diff
@@ -0,0 +1,204 @@
+import styled from 'styled-components'
+import { useCallback, useEffect, useRef, useState } from 'react'
+import { colors } from '../style/colors'
+import { ReactComponent as WriteIcon } from '../assets/svgs/write.svg'
+import { ReactComponent as EditIcon } from '../assets/svgs/edit.svg'
+import { ReactComponent as FriendsIcon } from '../assets/svgs/friends.svg'
+import { ReactComponent as ChattingIcon } from '../assets/svgs/chatting.svg'
+import { ReactComponent as SettingIcon } from '../assets/svgs/setting.svg'
+import { ReactComponent as SearchIcon } from '../assets/svgs/search.svg'
+import { FriendList } from './FriendList'
+import { ChatList } from './ChatList'
+import { MyPage } from './MyPage'
+
+export const TapBar = () => {
+  const inputRef = useRef<HTMLInputElement>(null)
+  //탭바 상태 localStorage에 저장
+  const [isFriend, setIsFriend] = useState<boolean>(
+    localStorage.getItem('isFriend') ? localStorage.getItem('isFriend') === 'true' : true,
+  )
+  const [isChatList, setIsChatting] = useState<boolean>(
+    localStorage.getItem('isChatList') ? localStorage.getItem('isChatList') === 'true' : false,
+  )
+  const [isSetting, setIsSetting] = useState<boolean>(
+    localStorage.getItem('isSetting') ? localStorage.getItem('isSetting') === 'true' : false,
+  )
```

Comment:
로직이 반복되고 있어서 이 부분도 key 를 받아서 해당 키를 가진 localStorage 데이터가 있으면 true, 아니면 false 를 반환하는 메소드를 하나 생성하여 가져다 쓰면 더 효율적인 코드가 될 것 같아요 ㅎㅎ

```suggestion

const getTabBarState=(key:string)=>{
      if(localStorage.getItem(key)) return true;
      return false
}

const [isFriend, setIsFriend] = useState<boolean>(getTabBarState('isFriend'))
const [isChatList, setIsChatList] = useState<boolean>(getTabBarState('isChatList'))
const [isSetting, setIsSetting] = useState<boolean>(getTabBarState('isSetting'))

Reply:
오 생각을 못했는데 감사합니다~~!